doc/nrf/protocols/matter/getting_started/memory_profiling.rst

```diff
@@ -0,0 +1,371 @@
+
+The :ref:`ug_matter_configuring_settings_shell` provides only the peak value of the current settings usage.
+To estimate the needed space for the ``settings_storage`` partitions you shall gather size of each settings key and decide how often the value is updated during the device's lifetime.
+
+
+
+Stack usage for all threads
+---------------------------
+
+Requirements:
+
+.. parsed-literal::
+   :class: highlight
+
+    CONFIG_SHELL=y
+    CONFIG_KERNEL_SHELL=y
+
+You can also measure the peak measurement of the stack usage of each thread running on the Matter device.
+This measurement may help in setting the proper stack size value and saving RAM space for other stacks or the heap.
+
+To see all statistics for each running thread, use the following UART shell command on your device:
+
+.. code-block:: console
+
+    kernel stacks
+
+You will see the similar statistics as following ones, the number of threads may be different:
+
+.. code-block:: console
+
+    0x20011568 CHIP                             (real size 6144):	unused 3952	usage 2192 / 6144 (35 %)
+    0x200069e8 BT RX WQ                         (real size 1216):	unused 1040	usage  176 / 1216 (14 %)
+    0x20006930 BT TX                            (real size 1536):	unused 1080	usage  456 / 1536 (29 %)
+    0x20006d08 rx_q[0]                          (real size 1536):	unused 1384	usage  152 / 1536 ( 9 %)
+    0x20006e18 openthread                       (real size 4096):	unused 3432	usage  664 / 4096 (16 %)
+    0x20007be8 ot_radio_workq                   (real size 1024):	unused  848	usage  176 / 1024 (17 %)
+    0x20006768 shell_uart                       (real size 3200):	unused 2104	usage 1096 / 3200 (34 %)
+    0x20002580 nrf5_rx                          (real size 1024):	unused  832	usage  192 / 1024 (18 %)
+    0x2000d510 sysworkq                         (real size 1152):	unused  880	usage  272 / 1152 (23 %)
+    0x20007b10 MPSL Work                        (real size 1024):	unused  808	usage  216 / 1024 (21 %)
+    0x2000d3a0 idle                             (real size  320):	unused  272	usage   48 /  320 (15 %)
+    0x2000d458 main                             (real size 6144):	unused 4584	usage 1560 / 6144 (25 %)
+    0x20025d00 IRQ 00                           (real size 2048):	unused 1120	usage  928 / 2048 (45 %)
+
+
+You can read the peak usage measurement for each thread and to learn about the total size of the stack, and unused bytes.
+Using these estimations you can adjust the stack values for your application.
+
+Configuring memory usage
+========================
+
+Most of the Matter samples in |NCS| have a safe configuration that assumes a high number of free space for heap, stacks, and settings partition sizes.
+After measuring the memory usage, you may want to adjust the memory parameters according to your project requirements.
+
+The following sections present a guide on how to adjust specific maximum memory values.
+
+Settings usage
```

Comment:
If we want to describe it here, remember we have to set `CONFIG_SETTINGS_NVS_SECTOR_COUNT` in the consistent way. Also note that we have similar information mentioned (not that detailed) in `ug_matter_device_bootloader_partition_layout`, so maybe it would be good to extend it there and only put ref link here, or something.